process/transaction/process.go

```diff
@@ -75,23 +92,101 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction, round
 		return err
 	}
 
-	txType, err := txProc.scProcessor.ComputeTransactionType(tx)
+	txType, err := txProc.txTypeHandler.ComputeTransactionType(tx)
 	if err != nil {
 		return err
 	}
 
 	switch txType {
 	case process.MoveBalance:
-		return txProc.processMoveBalance(tx, adrSrc, adrDst)
+		currTx, ok := tx.(*transaction.Transaction)
+		if !ok {
+			return process.ErrWrongTypeAssertion
+		}
+		return txProc.processMoveBalance(currTx, adrSrc, adrDst)
 	case process.SCDeployment:
-		return txProc.processSCDeployment(tx, adrSrc, roundIndex)
+		currTx, ok := tx.(*transaction.Transaction)
+		if !ok {
+			return process.ErrWrongTypeAssertion
+		}
+		return txProc.processSCDeployment(currTx, adrSrc, roundIndex)
 	case process.SCInvoking:
-		return txProc.processSCInvoking(tx, adrSrc, adrDst, roundIndex)
+		currTx, ok := tx.(*transaction.Transaction)
+		if !ok {
+			return process.ErrWrongTypeAssertion
+		}
+		return txProc.processSCInvoking(currTx, adrSrc, adrDst, roundIndex)
+	case process.TxFee:
+		currTxFee, ok := tx.(*feeTx.FeeTx)
+		if !ok {
+			return process.ErrWrongTypeAssertion
+		}
+		return txProc.processAccumulatedTxFees(currTxFee, adrSrc)
 	}
 
 	return process.ErrWrongTransaction
 }
 
+func (txProc *txProcessor) processTxFee(tx *transaction.Transaction, acntSnd *state.Account) (*feeTx.FeeTx, error) {
+	if acntSnd == nil {
+		return nil, nil
+	}
+
+	cost := big.NewInt(0)
+	cost = cost.Mul(big.NewInt(0).SetUint64(tx.GasPrice), big.NewInt(0).SetUint64(tx.GasLimit))
+
+	txDataLen := int64(len(tx.Data)) + 1
+	minFee := big.NewInt(0)
+	minFee = minFee.Mul(big.NewInt(txDataLen), big.NewInt(MinGasPrice))
+	minFee = minFee.Add(minFee, big.NewInt(MinTxFee))
```

Comment:
Shouldn't `MinTxFee` be the fee value for the case the calculation before is less than the minimum?

Reply:
If you add Data into the transaction it costs more, as you are moving data as well.